File: frontend/src/services/commission.service.js
import api from "./axios.config.js";

// 1. Tạo commission
const createCommission = async (payload) => {
    try {
        const response = await api.post(`/commissions`, payload);
        return response.data;
    } catch (error) {
        console.error("Create Commission Error:", error);
        throw error;
    }
};

// 2. Xóa/Lưu trữ commission
const deleteCommission = async (id, payload) => {
    try {
        const response = await api.delete(`/commissions/${id}`, { data: payload });
        return response.data;
    } catch (error) {
        console.error("Delete Commission Error:", error);
        throw error;
    }
};

// 3. Lấy danh sách tất cả commission
const getAllCommissions = async () => {
    try {
        const response = await api.get(`/commissions`);
        return response.data;
    } catch (error) {
        console.error("Get All Commissions Error:", error);
        throw error;
    }
};

// 4. Lấy chi tiết commission
const getCommissionDetail = async (id) => {
    try {
        const response = await api.get(`/commissions/${id}`);
        return response.data;
    } catch (error) {
        console.error("Get Commission Detail Error:", error);
        throw error;
    }
};

// 5. Cập nhật commission
const updateCommission = async (payload) => {
    try {
        const response = await api.put(`/commissions`, payload);
        return response.data;
    } catch (error) {
        console.error("Update Commission Error:", error);
        throw error;
    }
};

// 6. Kích hoạt DEAL commission
const activeCommission = async (commissionId) => {
    try {
        const response = await api.patch(`/commissions/active/${commissionId}`);
        return response.data;
    } catch (error) {
        console.error("Active Commission Error:", error);
        throw error;
    }
};

// 7. Lấy số lượng và danh sách hotel đang sử dụng DEAL
const getHotelsUsingDeal = async (commissionId) => {
    try {
        const response = await api.get(`/commissions/hotels/${commissionId}`);
        return response.data;
    } catch (error) {
        console.error("Get Hotels Using Deal Error:", error);
        throw error;
    }
};

// 8. Reset commission của hotel về mặc định
const setDefaultCommission = async (hotelId) => {
    try {
        const response = await api.patch(`/commissions/set-default/${hotelId}`);
        return response.data;
    } catch (error) {
        console.error("Set Default Commission Error:", error);
        throw error;
    }
};

// 9. Admin Lấy tất cả logs lịch sử commission
const getAllLogs = async () => {
    try {
        const response = await api.get(`/commissions/logs`);
        return response.data;
    } catch (error) {
        console.error("Get All Logs Error:", error);
        throw error;
    }
};

// 10. Hotel Lấy logs lịch sử của khách sạn hiện tại
const getMyHotelLogs = async () => {
    try {
        const response = await api.get(`/commissions/logs/hotel`);
        return response.data;
    } catch (error) {
        console.error("Get My Hotel Logs Error:", error);
        throw error;
    }
};

export const commissionService = {
    createCommission,
    deleteCommission,
    getAllCommissions,
    getCommissionDetail,
    updateCommission,
    activeCommission,
    getHotelsUsingDeal,
    setDefaultCommission,
    getAllLogs,
    getMyHotelLogs,
};

File: backend/src/main/java/com/HTPj/htpj/service/CommissionService.java
package com.HTPj.htpj.service;

import com.HTPj.htpj.dto.request.commission.CreateCommissionRequest;
import com.HTPj.htpj.dto.request.commission.DeleteCommissionRequest;
import com.HTPj.htpj.dto.request.commission.UpdateCommissionRequest;
import com.HTPj.htpj.dto.response.commision.CommissionDetailResponse;
import com.HTPj.htpj.dto.response.commision.CommissionLogResponse;
import com.HTPj.htpj.dto.response.commision.CommissionResponse;
import com.HTPj.htpj.dto.response.commision.HotelUsingDealResponse;

import java.util.List;

public interface CommissionService {
    String create(CreateCommissionRequest request);

    String delete(Long commissionId, DeleteCommissionRequest request);

    List<CommissionResponse> getAll();

    CommissionDetailResponse getDetail(Long commissionId);

    String update(UpdateCommissionRequest request);
    //active case deal
    String activeCommission(Long commissionId);

    //count hotel use deal
    HotelUsingDealResponse getHotelsUsingDeal(Long commissionId);

    //set to default:
    String setDefaultCommission(Integer hotelId);

    //admin
    List<CommissionLogResponse> getAllCommissionLogs();

    //hotel
    List<CommissionLogResponse> getHotelCommissionLogs();
}
